### PR TITLE
AA-277: Add certificate available alert on outline

### PR DESCRIPTION
**Pull request by mikix**
### Old:

Screenshot from 2020-08-03 12-07-38

### New:

Screenshot from 2020-08-03 12-07-43

### CODE REVIEW

File: src/alerts/certificate-available-alert/hooks.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const {
+    datesWidget: {
+      courseDateBlocks,
+      userTimezone,
+    },
+  } = useModel('outline', courseId);
+  const { username } = getAuthenticatedUser();
+
+  const certBlock = courseDateBlocks.find(b => b.dateType === 'certificate-available-date');
+  const endBlock = courseDateBlocks.find(b => b.dateType === 'course-end-date');
```
Comment (Dillon-Dumesnil) [inline src/alerts/certificate-available-alert/hooks.js:22]: Is this how we should be finding this information? Or are we just using the fact that these are also returned in the outline to our advantage?

File: src/alerts/certificate-available-alert/CertificateAvailableAlert.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  } = payload;
+
+  const timezoneFormatArgs = userTimezone ? { timeZone: userTimezone } : {};
+
+  return (
+    <Alert type={ALERT_TYPES.INFO}>
+      <strong>
+        <FormattedMessage
+          id="learning.outline.alert.cert.title"
+          defaultMessage="We are working on generating course certificates."
```
Comment (Dillon-Dumesnil) [inline src/alerts/certificate-available-alert/CertificateAvailableAlert.jsx:23]: Should all these messages be internationalized?

**Review by Dillon-Dumesnil — COMMENTED** [review]
Couple questions. As a more general question, should these alerts maybe live under the Course Home folder? `src/course-home/alerts/...`?

File: src/alerts/certificate-available-alert/hooks.js (hunk 1 above)
Comment (mikix) [reply src/alerts/certificate-available-alert/hooks.js:22]: The LMS alerts piggy back off the date blocks too, so I was kind of reproducing that logic here (at least for the cert available date).

For the course end date... mostly just using the fact that it's already there in the course date blocks to my advantage.

**Comment by mikix** [comment]
> Couple questions. As a more general question, should these alerts maybe live under the Course Home folder? src/course-home/alerts/...?

Possibly...?  These alerts are scoped to the outline tab (topic of "course-outline-alerts" for example), so they can only reasonably be used there. So maybe they should live there.

But some alerts will eventually be shared? And I was frankly being lazy in not thinking about how they should all be organized.

**Review by Dillon-Dumesnil — APPROVED** [review]

File: src/alerts/certificate-available-alert/CertificateAvailableAlert.jsx (hunk 2 above)
Comment (Dillon-Dumesnil) [reply src/alerts/certificate-available-alert/CertificateAvailableAlert.jsx:23]: Didn't realize FormattedMessage is for internationalization

**Pull request merged**